src/aws.js

```diff
@@ -2,6 +2,32 @@ const AWS = require('aws-sdk');
 const core = require('@actions/core');
 const config = require('./config');
 
+// User data scripts are run as the root user
+function buildUserDataScript(githubRegistrationToken, label) {
+  if (config.input.runnerHomeDir) {
+    // If runner home directory is specified, we expect the actions-runner software (and dependencies)
+    // to be pre-installed in the AMI, so we simply cd into that directory and then start the runner
+    return [
+      '#!/bin/bash',
+      `cd "${config.input.runnerHomeDir}"`,
+      'export RUNNER_ALLOW_RUNASROOT=1',
+      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
+      './run.sh',
+    ];
+  } else {
+    return [
+      '#!/bin/bash',
+      'mkdir actions-runner && cd actions-runner',
+      'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
+      'curl -O -L https://github.com/actions/runner/releases/download/v2.286.0/actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
+      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
+      'export RUNNER_ALLOW_RUNASROOT=1',
+      `./config.sh --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
+      './run.sh',
+    ];
+  }
+}
+
 async function startEc2Instance(label, githubRegistrationToken) {
   const ec2 = new AWS.EC2();
 
```

Comment:
place 1 of merge conflict merging with upstream: 
the original wanted to use  `const userData = buildUserDataScript(...)` 
I kept the current userData

Reply:
The function should be fine to switch to, we just need to make sure the user data contents have what we need.

I would use this resolution:
```
// User data scripts are run as the root user
function buildUserDataScript(githubRegistrationToken, label) {
  if (config.input.runnerHomeDir) {
    // If runner home directory is specified, we expect the actions-runner software (and dependencies)
    // to be pre-installed in the AMI, so we simply cd into that directory and then start the runner
    return [
      '#!/bin/bash',
      `cd "${config.input.runnerHomeDir}"`,
      'export RUNNER_ALLOW_RUNASROOT=1',
      'sudo echo "export RUNNER_ALLOW_RUNASROOT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false" >> /etc/profile.d/env.sh',
      `./config.sh --unattended --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
      './run.sh',
    ];
  } else {
    return [
      '#!/bin/bash',
      'mkdir actions-runner && cd actions-runner',
      'case $(uname -m) in aarch64) ARCH="arm64" ;; amd64|x86_64) ARCH="x64" ;; esac && export RUNNER_ARCH=${ARCH}',
      'curl -O -L https://github.com/actions/runner/releases/download/v2.286.0/actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
      'tar xzf ./actions-runner-linux-${RUNNER_ARCH}-2.286.0.tar.gz',
      'export RUNNER_ALLOW_RUNASROOT=1',
      'sudo echo "export RUNNER_ALLOW_RUNASROOT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1" >> /etc/profile.d/env.sh',
      'export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false',
      'sudo echo "export DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY=false" >> /etc/profile.d/env.sh',
      `./config.sh --unattended --url https://github.com/${config.githubContext.owner}/${config.githubContext.repo} --token ${githubRegistrationToken} --labels ${label}`,
      './run.sh',
    ];
  }
}
```

Reply:
switched

Reply:
`config` doesn't set `runnerHomeDir` right now so I think it's always going to be the `else` clause